TRX/Tron-overview.md

```diff
@@ -183,25 +183,29 @@ try {
 # 9. Calculation of block ID
 Block ID is a combination of block height and the hash of the blockheader’s raw data. To get block ID, first hash the raw data of the blockheader and replace the first 8 bytes of the hash with the blockheight, as the following:
 ```
-private byte[] generateBlockId(long blockNum, byte[] blockHash) { 
- byte[] numBytes = Longs.toByteArray(blockNum); 
- byte[] hash = blockHash; 
- System.arraycopy(numBytes, 0, hash, 0, 8); 
+private byte[] generateBlockId(long blockNum, byte[] blockHash) {
+ byte[] numBytes = Longs.toByteArray(blockNum);
+ byte[] hash = blockHash;
+ System.arraycopy(numBytes, 0, hash, 0, 8);
  return hash;
- }
+}
 ```
 BlockHash is the hash of the raw data of the blockheader, which can be calculated as the following:
 ```
 Sha256Hash.of(this.block.getBlockHeader().getRawData().toByteArray())
 ```
 # 10. Construction and signature of transaction
-There are two ways to construct a transaction:
+Based on your own needs, there are two ways to construct the transaction: invoking API on the full node or filling in every fields of a transaction manually at local.
+
 ## 10.1 Invoke APIs on the full node
-Based on your own needs, construct a corresponding local Contract and construct transactions with corresponding APIs. For the contract, please refer to https://github.com/TRONprotocol/protocol/blob/master/core/Contract.proto.
+You could construct transactions with corresponding APIs.
+
+- gRPC API: https://github.com/tronprotocol/Documentation/blob/master/English_Documentation/TRON_Protocol/TRON_Wallet_RPC-API.md
+- HTTP API: https://github.com/tronprotocol/Documentation/blob/master/TRX/Tron-http.md
 
```

Comment:
"Individual Contract protocol files are available here: https://github.com/TRONprotocol/protocol/blob/master/core/Contract.proto"

Reply:
this is "invoke API on full node" section. so I think that it doesn't need to mention Contract.proto file in here.

Reply:
ah. I agree. Checking types of transactions would be helpful for dev. I'll use Contract.proto link instead of APIs links.

Reply:
I think the API links are still useful as you can construct a transaction object through the node API (you just need the contract proto files for adding the different contracts to the transaction)

Reply:
then I'll add your suggested sentences after HTTP API link.

> "Individual Contract protocol files are available here: https://github.com/TRONprotocol/protocol/blob/master/core/Contract.proto"

Reply:
Looks good